ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/anomalylocalization/AnomalyLocalizerImpl.java

```diff
@@ -126,6 +127,10 @@ private void onOverallAggregatesResponse(MultiSearchResponse response, AnomalyLo
         AnomalyLocalizationOutput.Result result = new AnomalyLocalizationOutput.Result();
         List<Map.Entry<Long, Long>> intervals = timeBuckets.getAllIntervals();
         for (int i = 0; i < intervals.size(); i++) {
+            if (response.getResponses()[i].getFailure() instanceof IndexNotFoundException) {
```

Comment:
Suggest check if index exists or not like this https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/indices/MLIndicesHandler.java#L90